pandas/src/parser/tokenizer.c

```diff
@@ -679,7 +680,16 @@ static int parser_buffer_bytes(parser_t *self, size_t nbytes) {
     }
 
 int skip_this_line(parser_t *self, int64_t rownum) {
-    if (self->skipset != NULL) {
+    int should_skip;
+    PyGILState_STATE state;
+
+    if (self->skipfunc != NULL) {
+        state = PyGILState_Ensure();
+        should_skip = PyObject_IsTrue(PyObject_CallFunction(
```

Comment:
Ah, good catch (in so many senses of the word :smile:).  Turns out you caught another bug with this, as attempts to trap this error revealed a bug in `raise_parser_error` in which we were assuming `PyErr_Fetch` always returned an `Exception`-type for `*value`, but that was not the case.